### ISSUE TITLE
After model.fit(para default), the model.predict() uses the last updated model weights to perform predicting?

### ISSUE THREAD

**Issue by Eadith** [keras-team/keras#14124]
When the model.fit() performed, I didn't save model, then using predict(), the weight used in predict() is the last epoch in fit() or not? Or the fit() will auto save a best model to perform predict()?

**Comment by Vincent9797** [comment]
Yes, it uses the weight from the last epoch. If you want to use the best weights, use `SaveModelCallback()` where you can set `save_best_only=True` and you can define the best model based on any metrics included in .compile or just simply the loss function.

After training, you should be able to see the weights file saved based on the file path defined by you when you instantiate that callback. Load the model with those weights and you can perform `predict()` with the best weights after that.

**Comment by sushreebarsa** [comment]
@Eadith Moving this issue to closed status as there has been no recent activity, in case you still face the error please create a new issue.Thanks!
